### PR TITLE
Refactor Force based data sets

### LINKED ISSUE #139: Refactor FSI Stress/Force
As observed in #133 and discussed in #134 , the current design and implementation of writing forces and stresses is rather bad: The stress field inherits common functionality from the force field. Though the Force field creates its own data field, which is not required in case stress data is used. 

A temporary solution is given in #134. A much simpler and cleaner solution would be to create a force calculation base class for both fields, so that common functionality can be inherited independently in stress and force classes respectively.

### PR DESCRIPTION
**Pull request by DavidSCN**
I took the opportunity to clean-up the Force based data classes and split the common functionality in a base class. The PR reverts a crime introduced in #134 and partially #125, reduces code duplication and makes the whole data set much easier to maintain. Closes #139.

I hope we can run the tests on this PR.

### CODE REVIEW

File: FSI/Stress.C [hunk 1]
```diff
@@ -49,96 +32,17 @@ solverType_(solverType)
...
-    */
-    FatalErrorInFunction
-        << "Reading stresses is not supported."
-        << exit(FatalError);
+    this->readFromBuffer(buffer);
+}
+
+vectorField preciceAdapter::FSI::Stress::getFaceVectors(const unsigned int patchID) const
+{
+  return mesh_.boundary()[patchID].nf();
```
Comment (MakisH) [inline FSI/Stress.C:45]: Maybe add a comment here on what `nf()` does, similar to the one for `Sf()`?

**Review by MakisH — APPROVED** [review]
Looking at the code, I like this refactoring. :+1:

- Before we had `Force` and `Stress : public Force`.
- Now we have a `ForceBase` with most functionality, and `Force` and `Stress` essentially changing the normal vectors they are using, overriding the following virtual function:

```c++
Foam::vectorField getFaceVectors(const unsigned int patchID) const override;
```
In the case of Force:
```c++
  // Normal vectors multiplied by face area
  return mesh_.boundary()[patchID].Sf();
  ```
In the case of Stress:
```c++
return mesh_.boundary()[patchID].nf();
```

This PR removes more code than adds, so this is highly encouraging! I have not tried to build or test the code, but if you already checked 1-2 cases, it should be fine.

**Comment by DavidSCN** [comment]
> This PR removes more code than adds, so this is highly encouraging! I have not tried to build or test the code, but if you already checked 1-2 cases, it should be fine.

Yes. Considering that there is now more 'c++ structure' code, it removes even more code tom maintain than shown in the overview. Let me address your comment and check the result again.

File: FSI/Stress.C (hunk 1 above)
Comment (DavidSCN) [reply FSI/Stress.C]: Done.

**Pull request merged**